content/blog/aws-eks-auto-mode/index.md

```diff
@@ -0,0 +1,2059 @@
+
+Common tasks that previously required significant expertise and time are now managed by AWS, making it simpler to run production-grade Kubernetes workloads. This includes:
+
+- Setting up and maintaining cluster autoscaling
+- Configuring and updating networking plugins
+- Managing storage drivers and volume provisioning
+- Handling node updates and security patches
+- Optimizing instance selection and placement
+
+## Pulumi Support for EKS Auto Mode
+
+We have added EKS Auto Mode support across our AWS provider ecosystem:
+
+- Pulumi AWS Provider (v6.63.0+)
+- Pulumi AWS Cloud Control Provider (v1.13.0+)
+- Pulumi EKS Provider (TODO: upcoming release)
+
+## Creating an EKS Auto Mode Cluster
+
+Let's explore how to leverage EKS Auto Mode with Pulumi by deploying an example.
+You can enable EKS Auto Mode by setting `autoMode.enabled` to true.
+
+First, we'll create a new EKS cluster with Auto Mode enabled. This requires properly configured networking and authentication settings:
+
+- VPC subnets are tagged appropriately for EKS Auto Mode load balancer discovery
+- Authentication Mode supports Access Entries (`Api` or `ApiAndConfigMap` mode) as required by EKS Auto Mode
+- Default node groups and security groups are skipped as Auto Mode manages these
+
+{{< chooser language "typescript,python,go,csharp,java,yaml" >}}
+
+{{% choosable language typescript %}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as awsx from "@pulumi/awsx";
+import * as eks from "@pulumi/eks";
+import * as k8s from "@pulumi/kubernetes";
+import { SubnetType } from "@pulumi/awsx/ec2";
+
+const config = new pulumi.Config();
+const clusterName = config.require("clusterName");
+
+const eksVpc = new awsx.ec2.Vpc("eks-auto-mode", {
+    enableDnsHostnames: true,
+    cidrBlock: "10.0.0.0/16",
+    subnetSpecs: [
+        // Necessary tags for EKS Auto Mode to identify the subnets for the load balancers.
+        // See: https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.1/deploy/subnet_discovery/
+        {type: SubnetType.Public, tags: {[`kubernetes.io/cluster/${clusterName}`]: "shared", "kubernetes.io/role/elb": "1"}},
+        {type: SubnetType.Private, tags: {[`kubernetes.io/cluster/${clusterName}`]: "shared", "kubernetes.io/role/internal-elb": "1"}},
+    ],
+    subnetStrategy: "Auto"
+});
+
+const cluster = new eks.Cluster("eks-auto-mode", {
+    name: clusterName,
+    // EKS Auto Mode requires Access Entries, use either the `Api` or `ApiAndConfigMap` authentication mode.
+    authenticationMode: eks.AuthenticationMode.Api,
+    skipDefaultNodeGroup: true,
```

Comment:
Should we remove these 2 defaults from the example?

Reply:
Good point, missed updating the examples after updating the PR!